sdk/eventhub/event-hubs/src/receiver.ts

```diff
@@ -160,25 +167,53 @@ export class EventHubConsumer {
   receive(onMessage: OnMessage, onError: OnError, abortSignal?: AbortSignalLike): ReceiveHandler {
     this._throwIfReceiverOrConnectionClosed();
     this._throwIfAlreadyReceiving();
+    const baseConsumer = this._baseConsumer!;
+
     if (typeof onMessage !== "function") {
       throw new TypeError("The parameter 'onMessage' must be of type 'function'.");
     }
     if (typeof onError !== "function") {
       throw new TypeError("The parameter 'onError' must be of type 'function'.");
     }
-    const checkpoint = this.getCheckpoint();
-    if (checkpoint) {
-      this._eventPosition = EventPosition.fromSequenceNumber(checkpoint);
+
+    // return immediately if the abortSignal is already aborted.
+    if (abortSignal) {
+      if (abortSignal.aborted) {
+        onError(new AbortError("The receive operation has been cancelled by the user."));
+        // close this receiver when user triggers a cancellation.
+        this.close().catch(() => {}); // no-op close error handler
+        return new ReceiveHandler(baseConsumer);
+      }
+
+      abortSignal.addEventListener("abort", () => {
+        if (this._baseConsumer) {
+          this._baseConsumer.abort();
+        }
+      });
     }
-    this._streamingReceiver = StreamingReceiver.create(
-      this._context,
-      this.consumerGroup,
-      this.partitionId,
-      this._eventPosition,
-      this._receiverOptions
+
+    const wrappedOnError = (error: Error) => {
+      // ignore retryable errors
+      if ((error as MessagingError).retryable) {
+        return;
+      }
+
+      if (error.name === "AbortError") {
+        // close this receiver when user triggers a cancellation.
+        this.close().catch(() => {}); // no-op close error handler
```

Comment:
Don't we handle closing in case of abort from the inside?

Reply:
Also we have the issue #4314 to look at closing links on abort. Can we keep these changes out of this PR?

Reply:
This is due to the new requirement we decided where the consumer should be permanently closed when an operation is cancelled, not just the underlying link.